src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace NServiceBus.Transports.RabbitMQ.Connection
+{
+    using System;
+    using global::RabbitMQ.Client;
+    using NServiceBus.Transports.RabbitMQ.Config;
+
+    class RabbitMqConnectionFactory
+    {
+        public ConnectionConfiguration Configuration { get; private set; }
+
+        public RabbitMqConnectionFactory(ConnectionConfiguration connectionConfiguration)
+        {
+            if (connectionConfiguration == null)
+            {
+                throw new ArgumentNullException("connectionConfiguration");
```

Comment:
why r we arg checking in an internal class? shouldnt this be verified at the point of the public API?